c++/src/H5DataSet.cpp

```diff
@@ -759,7 +758,7 @@ DataSet::p_read_variable_len(const hid_t mem_type_id, const hid_t mem_space_id,
 
     // Get string from the C char* and release resource allocated by C API
     strg = strg_C;
-    HDfree(strg_C);
+    free(strg_C);
```

Comment:
Should this be delete?

Reply:
No. We get that memory from the HDF5 C library, so it wasn't allocated with new.

Reply:
Actually, I remember why I used HDfree, because of the very reason you said here.  The right fix is to remove "new char[data_size + 1]();" above.  The library will allocate for strg_C.

Reply:
That new char[data_size + 1] line is in a different function. In this function, strg_C is just defined as a char pointer and not initialized.